Gemfile

```diff
@@ -1,16 +1,26 @@
 source 'https://rubygems.org'
 
-gemspec
-
 group :development do
-  gem 'vagrant',      git: 'https://github.com/mitchellh/vagrant.git'
-  gem 'rake'
+  gem 'vagrant',
+      :git => 'git://github.com/mitchellh/vagrant.git',
+      :ref => 'v1.8.1'
+
+  # test dependencies
+  gem 'minitest'
+  gem 'mocha'
+
   gem 'cucumber', '~> 2.1'
   gem 'aruba', '~> 0.13'
   gem 'komenda', '~> 0.1.6'
   gem 'launchy'
   gem 'gem-compare'
   gem 'mechanize'
+
+  # build tool
+  gem 'rake', '10.4.2'
```

Comment:
> And TBH, locking version of Rake due to some warnings is a bit shortsighted because:
> Its just warnings of test suite, so although they might be annoying, they makes no real harm

If I run the test I want a clean output. I don't want to see warnings. And in this case, even though the tests will eventually run, the warning seem to indicate that there is some recursive loading of files going on. I'd rather avoid that.

Also if I run the tests in my IDE it will fill up my console with warnings to an end that the IDE reports issues.

> You will never discover that the warnings were fixed, since you locked the version and using the old version

Rake is a development dependency. I use what makes sense. There is no need to "discover" updates. As long as the used version does what it should, all is good. Nothing stops you to once in a while try to test newer versions, but that can be done by a single developer. Different story with a runtime dependency, where you are more interested in getting the latest in order to get potential bug fixes and improvements. That said, even there I prefer to lock versions and do upgrades explicitly.